lib/validators/specResolver.ts

```diff
@@ -931,6 +933,10 @@ export class SpecResolver {
         if (d.$ref) {
           delete d.$ref
         }
+        const xMsEnum = d["x-ms-enum"]
+        if (xMsEnum !== undefined) {
+          ps.asMutable(xMsEnum).modelAsString = false
```

Comment:
i would say explain why we do this

Reply:
there are cases were this should be true, like on enums not part of discriminator types, how are we affecting those cases? we generally recommend modelAsString= true for specs, to avoid breaking changes in future api-versions for code generation, so I'd like to understand more of what's not working due to this value for the tool